aiida/restapi/translator/nodes/data/upf.py

```diff
@@ -33,44 +32,13 @@ class UpfDataTranslator(DataTranslator):
 
     _result_type = __label__
 
-    def __init__(self, **kwargs):
-        """
-        Initialise the parameters.
-        Create the basic query_help
-        """
-        super(UpfDataTranslator, self).__init__(Class=self.__class__, **kwargs)
-
-    @staticmethod
-    def get_visualization_data(node, visformat=None):
-        """
-
-        Returns: data in a format required by dr.js to visualize a 2D plot
-        with multiple data series.
-
-        """
-        return []
-
     @staticmethod
-    def get_downloadable_data(node, download_format=None):
+    def get_derived_properties(node):
         """
-        Generic function extended for upf data
-
-        :param node: node object that has to be downloaded
-        :param download_format: file extension format
-        :returns: data in selected format to download
+        :param node: node object
+        :returns: raise RestFeatureNotAvailable exception
         """
 
-        response = {}
-
-        filename = node.filename
-        try:
-            content = node.get_content()
-        except IOError:
-            error = 'Error in getting {} content'.format(filename)
-            raise RestInputValidationError(error)
-
-        response['status'] = 200
-        response['data'] = content
-        response['filename'] = filename
+        from aiida.restapi.common.exceptions import RestFeatureNotAvailable
 
-        return response
+        raise RestFeatureNotAvailable('This endpoint is not available for upf.')
```

Comment:
Here, and for others: if there is not derived property, shouldn't we just return an empty dict rather than raising?

Reply:
Changed to return an empty dict.

Reply:
Is this done for all other data types?

Reply:
Yes, for all other data types and process types too.